src/dysh/spectra/spectrum.py

```diff
@@ -572,11 +572,11 @@ def __add__(self, other):
         return addition
 
     def __sub__(self, other):
-        subtraction = self.sub(other, **{"handle_meta": self._add_meta})
+        subtraction = self.subtract(other, **{"handle_meta": self._add_meta})
         subtraction._target = self._target
         return subtraction
 
-    def add_meta(self, operand, operand2=None, **kwargs):
+    def _add_meta(self, operand, operand2=None, **kwargs):
         meta = deepcopy(operand)
         meta["EXPOSURE"] = operand["EXPOSURE"] + operand2["EXPOSURE"]
```

Comment:
Why does operand2 default to None instead of uninitialized?   If it remains None, line 581 will raise an exception.

Reply:
Good point. I was following the advice from `astropy` [here].(https://github.com/astropy/astropy/blob/75ce25ee3718cef767ca41dcbbf724a28bbe60a5/astropy/nddata/mixins/ndarithmetic.py#L593)

Reply:
that astropy call goes thru [self._prepare_then_do_arithmetic() ](https://github.com/astropy/astropy/blob/75ce25ee3718cef767ca41dcbbf724a28bbe60a5/astropy/nddata/mixins/ndarithmetic.py#L657l) which checks for the None.  I think just add 

    if operand2 is None:
        meta["EXPOSURE"] = operand["EXPOSURE"]